backend/api/Services/DeckService.cs

```diff
@@ -0,0 +1,144 @@
+﻿using Api.Controllers.Models;
+using Api.Database.Context;
+using Api.Database.Models;
+using Api.Utilities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Api.Services
+{
+    public interface IDeckService
+    {
+        public abstract Task<IEnumerable<Deck>> ReadAll();
+
+        public abstract Task<Deck?> ReadById(string id);
+
+        public abstract Task<IEnumerable<Deck>> ReadByInstallation(string installationCode);
+
+        public abstract Task<Deck?> ReadByName(string deckName);
+
+        public abstract Task<Deck?> ReadByInstallationAndPlantAndName(Installation installation, Plant plant, string deckName);
+
+        public abstract Task<Deck> Create(CreateDeckQuery newDeck);
+
+        public abstract Task<Deck> Update(Deck deck);
+
+        public abstract Task<Deck?> Delete(string id);
+
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1309:Use ordinal StringComparison",
+        Justification = "EF Core refrains from translating string comparison overloads to SQL"
+    )]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1304:Specify CultureInfo",
+        Justification = "Entity framework does not support translating culture info to SQL calls"
+    )]
+    public class DeckService : IDeckService
+    {
+        private readonly FlotillaDbContext _context;
+        private readonly IInstallationService _installationService;
+        private readonly IPlantService _plantService;
+
+        public DeckService(FlotillaDbContext context, IInstallationService installationService, IPlantService plantService)
+        {
+            _context = context;
+            _installationService = installationService;
+            _plantService = plantService;
+        }
+
+        public async Task<IEnumerable<Deck>> ReadAll()
+        {
+            return await GetDecks().ToListAsync();
+        }
+
+        private IQueryable<Deck> GetDecks()
```

Comment:
Does this crash if there are no decks? Since we don't have the "?" after Deck

Reply:
Since it's returning the query instead of the object itself I don't think it should be null. But I could be wrong.

Reply:
You should know more than me for sure ahah thanks!